src/test/6_MainRegistry.t.sol

```diff
@@ -1038,3 +1038,142 @@ contract PricingLogicTest is MainRegistryTest {
+            })
+        );
+        mainRegistry.addBaseCurrency(
+            MainRegistry.BaseCurrencyInformation({
+                baseCurrencyToUsdOracleUnit: uint64(10 ** Constants.oracleEthToUsdDecimals),
+                assetAddress: address(eth),
+                baseCurrencyToUsdOracle: address(oracleEthToUsd),
+                baseCurrencyLabel: "ETH",
+                baseCurrencyUnitCorrection: uint64(10 ** (18 - Constants.ethDecimals))
+            })
+        );
+        mainRegistry.addPricingModule(address(standardERC20PricingModule));
+        mainRegistry.addPricingModule(address(floorERC721PricingModule));
+        standardERC20PricingModule.addAsset(address(eth), oracleEthToUsdArr, emptyRiskVarInput, type(uint128).max);
+        standardERC20PricingModule.addAsset(address(link), oracleLinkToUsdArr, emptyRiskVarInput, type(uint128).max);
+
+        floorERC721PricingModule.addAsset(
+            address(bayc), 0, type(uint256).max, oracleWbaycToEthEthToUsd, emptyRiskVarInput, type(uint128).max
+        );
+
+        mainRegistry.setFactory(address(factory));
+
+        vm.stopPrank();
+
+        vm.startPrank(vaultOwner);
+        proxyAddr = factory.createVault(uint256(keccak256(abi.encodePacked(block.timestamp, msg.sender))), 1);
+        vm.stopPrank();
+    }
+
+    function testSuccess_batchProcessDeposit_directCall(uint128 amountLink) public {
+        address[] memory assetAddresses = new address[](1);
+        assetAddresses[0] = address(link);
+
+        uint256[] memory assetIds = new uint256[](1);
+        assetIds[0] = 0;
+
+        uint256[] memory assetAmounts = new uint256[](1);
+        assetAmounts[0] = amountLink;
+
+        (, uint128 oldExposure) = standardERC20PricingModule.exposure(address(link));
+
+        vm.startPrank(proxyAddr);
+        mainRegistry.batchProcessDeposit(assetAddresses, assetIds, assetAmounts);
+        vm.stopPrank();
+
+        (, uint128 newExposure) = standardERC20PricingModule.exposure(address(link));
+
+        assertEq(newExposure, oldExposure + amountLink);
+    }
+
+    function testRevert_batchProcessDeposit_delegateCall(uint128 amountLink) public {
+        address[] memory assetAddresses = new address[](1);
+        assetAddresses[0] = address(link);
+
+        uint256[] memory assetIds = new uint256[](1);
+        assetIds[0] = 0;
+
+        uint256[] memory assetAmounts = new uint256[](1);
+        assetAmounts[0] = amountLink;
+
+        vm.startPrank(proxyAddr);
+        vm.expectRevert("Delegate calls not allowed.");
+        (bool success,) = address(mainRegistry).delegatecall(
+            abi.encodeWithSignature(
+                "batchProcessDeposit(address[] calldata,uint256[] calldata,uint256[] calldata)",
+                assetAddresses,
+                assetIds,
+                assetAmounts
+            )
+        );
+        vm.stopPrank();
+
+        success; //avoid warning
```

Comment:
```suggestion
```

Reply:
will throw a 'return of delegatecall not evaluate' or something similar